bitcoin/src/crypto/taproot.rs

```diff
@@ -54,6 +58,24 @@ impl Signature {
         }
         ser_sig
     }
+
+    /// Serializes the signature (without heap allocation)
+    ///
+    /// This returns a type with an API very similar to that of `Box<[u8]>`.
+    /// You can get a slice from it using deref coercions or turn it into an iterator.
+    pub fn serialize(self) -> SerializedSignature {
+        let mut buf = [0; serialized_signature::MAX_LEN];
+        let ser_sig = self.sig.serialize();
+        buf[..64].copy_from_slice(&ser_sig);
+        let len = if self.hash_ty == TapSighashType::Default {
+            // default sighash type, don't add extra sighash byte
+            64
+        } else {
+            buf[64] = self.hash_ty as u8;
+            65
+        };
+        SerializedSignature::from_raw_parts(buf, len)
```

Comment:
To me 64 has different semantics than `MAX_LEN - 1`. The former is the length of serialized schnorr signature. But I guess it's splitting hair. I can use `MAX_LEN - 1` if that's preferred.

Reply:
Man I'm kinda leaning towards just having all magic numbers and not using `MAX_LEN`. I didn't write that yesterday because I'm usually the guy doing "remove magic number" patches.

Reply:
Ah, the buffer is not an arbitrary size so `MAX` is the wrong identifier, its a sig + sighash-byte length buffer.

Reply:
Will, it's still max, because it can hold shorter slice (without the sighash byte). It's just a weird case of having only two possible lengths. Also I think magic number would be fine but I'm not convinced I should go back and rewrite the PR. :)

Reply:
Yep, agree. I maintain my ACK :)